### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by joxl**
## Technical Summary

New PR template for this repository (demonstrated here!).

Ticket: [SAAS-13844](https://dimagi-dev.atlassian.net/browse/SAAS-13844)

### Safety story

PR process only, non-product.

### Automated test coverage

No tests.

### QA Plan

No QA.

### Special deploy instructions

- [x] This PR can be deployed after merge with no further considerations.

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations.

### Review
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]
Same comments as https://github.com/dimagi/commcare-core/pull/1146#pullrequestreview-1068903834

**Review by shubham1g5 — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### QA Plan
+<!--
+- Describe QA plan that along with automated test coverages proves this PR is regression free.
+- Link to QA Ticket
+-->
+
+### Migrations
+<!--
+Delete this section if the PR does not contain any migrations.
+See also: https://commcare-hq.readthedocs.io/migrations_in_practice.html
```
Comment (snopoke) [inline .github/PULL_REQUEST_TEMPLATE.md:35]: these docs don't really apply here

Comment (joxl) [reply .github/PULL_REQUEST_TEMPLATE.md:35]: Thanks @snopoke, I was a little conflicted on this myself. I looked around for "formplayer migration best practice" docs and couldn't find any and figured this doc was better than nothing (it offers _some_ generic migrations patterns).  Do you know if there's a migrations doc for formplayer that I can link in its place? If not I'll just remove this one without a replacement.

Comment (snopoke) [reply .github/PULL_REQUEST_TEMPLATE.md:35]: There isn't an equivalent doc. FP also does very few migrations of any kind.

**Comment by joxl** [comment]
PR description updated with latest version of the template.

**Review by snopoke — APPROVED** [review]

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**